lib/model.js

```diff
@@ -1287,7 +1386,49 @@ Model.prototype.all = function(options) {
  * ```sql
  * WHERE `Model`.`name` = 'a project' AND (`Model`.`id` IN (1, 2, 3) OR (`Model`.`id` > 10 AND `Model`.`id` < 100));
  * ```
- *
+ * __Typical CTE BFS down a tree to a certain depth__
+ * ```js
+ * 
+ * Model.findAll({
+ *  cte: [{
+ *      name: 'a',
+ *      cteAttributes: ['depth'],
+ *      initial: {
+ *          depth: 0
+ *          where: {
+ *              name: 'Jane'
+ *          }
+ *      },
+ *      recursive: {
+ *          depth: {
+ *              $add: [{$cte: 'depth'}, 1]
+ *          }
+ *          next: 'employees',
+ *          where: {
+ *              cte: {
+ *                  depth: { $lt: 4 }
+ *              }
+ *          }
+ *          
+ *      }
+ *  }],
+ *  cteSelect: 'a',
+ *  cteAttributes: ['depth']
+ * });
+ * ```
+ * ```sql
+ * WITH RECURSIVE a( `id`,`name`, `manager_id`,`depth` ) AS ( 
+ *   SELECT `model`.`id`, `model`.`name`, `model`.`manager_id`, 0 
+ *   FROM `models` AS `model` 
+ *   WHERE `model`.`name` = 'user2' 
```

Comment:
I don't have any preference, as long as the code and sql snippets match ;) - right now its Jane in the code and user2 in the sql